### PR TITLE
Deduplicate getting `TypeFunction` from `CallExp`

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

**Review by RazvanN7 — APPROVED** [review]

File: compiler/src/dmd/expression.d [hunk 1]
```diff
@@ -5329,6 +5329,26 @@ extern (C++) final class CallExp : UnaExp
     }
 }
 
+/**
+ * Get the called function type from a call expression
+ * Params:
+ *   ce = function call expression. Must have had semantic analysis done.
+ * Returns: called function type, or `null` if error / no semantic analysis done
+ */
+TypeFunction calledFunctionType(CallExp ce)
```
Comment (maxhaton) [inline compiler/src/dmd/expression.d:5338]:
```suggestion
TypeFunction calleeType(CallExp ce)
```

`const`/`inout` would be good here although I expect its not possible.

File: compiler/src/dmd/expression.d [hunk 2]
```diff
@@ -5329,6 +5329,26 @@ extern (C++) final class CallExp : UnaExp
     }
 }
 
+/**
+ * Get the called function type from a call expression
+ * Params:
+ *   ce = function call expression. Must have had semantic analysis done.
+ * Returns: called function type, or `null` if error / no semantic analysis done
```
Comment (maxhaton) [inline compiler/src/dmd/expression.d:5336]: Could it return `TypeError`?

File: compiler/src/dmd/expression.d (hunk 1 above)
Comment (dkorpel) [reply compiler/src/dmd/expression.d:5338]: Indeed can't call `toBasetype` on an `inout` object

Comment (maxhaton) [reply compiler/src/dmd/expression.d:5338]: The "called function" is the callee

Comment (maxhaton) [reply compiler/src/dmd/expression.d:5338]: You don't know the caller but if the CallExp has been resolved properly you always know the callee

**Pull request merged**